src/cache/file_cache.h

```diff
@@ -101,24 +102,55 @@ class FileCache {
                                   string::CommandLine command_line,
                                   string::Version version);
 
+  static string::HandledHash Hash(string::HandledSource code,
```

Comment:
I think changing the tests to reflect new signatures should not be hard :) But let's see what @abyss7 says.